code/SecurityAdmin.php

```diff
@@ -2,316 +2,66 @@
-            )->setTemplate('SilverStripe\\Forms\\CMSTabSet'),
-            // necessary for tree node selection in LeftAndMain.EditForm.js
-            new HiddenField('ID', false, 0)
-        );
-
-        // Add roles editing interface
-        $rolesTab = null;
-        if (Permission::check('APPLY_ROLES')) {
-            $rolesField = GridField::create(
-                'Roles',
-                false,
-                PermissionRole::get(),
-                GridFieldConfig_RecordEditor::create()
-            );
-
-            $rolesTab = $fields->findOrMakeTab('Root.Roles', _t(__CLASS__ . '.TABROLES', 'Roles'));
-            $rolesTab->push($rolesField);
+        $config = $gridField->getConfig();
+        /** @var GridFieldImportButton $importButton */
+        $importButton = $config->getComponentByType(GridFieldImportButton::class);
+        $name = strtolower($gridField->name);
+        if (str_contains($name, 'member')) {
+            $importButton
+                ->setImportIframe($memberImportController->Link('memberimport'))
+                ->setModalTitle(_t(__CLASS__ . '.IMPORTUSERS', 'Import users'));
+        } elseif (str_contains($name, 'group')) {
+            $importButton
+                ->setImportIframe($memberImportController->Link('groupimport'))
+                ->setModalTitle(_t(__CLASS__ . '.IMPORTGROUPS', 'Import groups'));
         }
-
-        // Build replacement form
-        $form = Form::create(
-            $this,
-            'EditForm',
-            $fields,
-            new FieldList()
-        )->setHTMLID('Form_EditForm');
-        $form->addExtraClass('cms-edit-form fill-height');
-        $form->setTemplate($this->getTemplatesWithSuffix('_EditForm'));
-        $form->addExtraClass('ss-tabset cms-tabset ' . $this->BaseCSSClasses());
-        $form->setAttribute('data-pjax-fragment', 'CurrentForm');
-
-        $this->extend('updateEditForm', $form);
-
-        return $form;
-    }
-
-    public function memberimport()
-    {
-        Requirements::clear();
-        Requirements::javascript('silverstripe/admin: client/dist/js/vendor.js');
-        Requirements::javascript('silverstripe/admin: client/dist/js/MemberImportForm.js');
-        Requirements::css('silverstripe/admin: client/dist/styles/bundle.css');
-
-        return $this->renderWith('BlankPage', [
-            'Form' => $this->MemberImportForm()->forTemplate(),
-            'Content' => ' '
-        ]);
-    }
-
-    /**
-     * @see SecurityAdmin_MemberImportForm
-     *
-     * @return Form
-     */
-    public function MemberImportForm()
-    {
-        if (!Permission::check('ADMIN')) {
-            return null;
+        // Limit import to admin since the import logic can affect assigned permissions
+        if (!Permission::check('ADMIN') || str_contains($name, 'role')) {
+            $config->removeComponentsByType(GridFieldImportButton::class);
         }
-
-        /** @var Group $group */
-        $group = $this->currentPage();
```

Comment:
I didn't copy this logic over MemberImportController because it always returned null, so it did nothing